Essentials/src/net/ess3/api/events/TPARequestEvent.java

```diff
@@ -0,0 +1,55 @@
+package net.ess3.api.events;
+
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import com.earth2me.essentials.CommandSource;
+
+import net.ess3.api.IUser;
+
+
+public class TPARequestEvent extends Event implements Cancellable {
+	private static final HandlerList handlers = new HandlerList();
+	private CommandSource requester;
+	private IUser requested;
+	private boolean cancelled = false, requestToTPAHere = false;
+
+	public TPARequestEvent(CommandSource requester, IUser requested, boolean tpaHere) {
+		super();
+		this.requester = requester;
+		this.requested = requested;
+		this.requestToTPAHere = tpaHere;
+	}
+
+	public CommandSource getRequester() {
+		return requester;
+	}
+
+	public IUser getRequested() {
+		return requested;
+	}
+	
+	public boolean isRequestToTeleportTo() {
```

Comment:
This method name is long and confusing. The `RequestTo` part is redundant since the whole event is regarding the event.

Reply:
Absolutely. Will be fixed in the next commit.